sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java

```diff
@@ -270,6 +273,68 @@ public void testReadWithWhere() throws Exception {
     pipeline.run();
   }
 
+  @Test
+  public void testSelectFields() throws Exception {
+    insertRecords();
+
+    PCollection<Scientist> output =
+        pipeline.apply(
+            CassandraIO.<Scientist>read()
+                .withHosts(pipeline.newProvider(Arrays.asList(CASSANDRA_HOST)))
+                .withPort(pipeline.newProvider(CASSANDRA_PORT))
+                .withKeyspace(pipeline.newProvider(CASSANDRA_KEYSPACE))
+                .withTable(pipeline.newProvider(CASSANDRA_TABLE))
+                .withSelectFields(pipeline.newProvider(Arrays.asList("person_id")))
+                .withCoder(SerializableCoder.of(Scientist.class))
+                .withEntity(Scientist.class)
+                .withWhere(pipeline.newProvider("person_id=10")));
+
+    PAssert.thatSingleton(output.apply("Count", Count.globally())).isEqualTo(1L);
+    PAssert.that(output)
+        .satisfies(
+            input -> {
+              for (Scientist sci : input) {
+                assertNotNull(sci.id);
+                assertNull(sci.name);
+              }
```

Comment:
What happens if you read `nameTs` here? Since objects are mapped I was wondering if it will be zero, probably worth asserting that to validate that the values were not 'filled' (given that they were not projected).

Reply:
nameTs will be null as it is not returned by Row so it's not mapped. 
Agree on assert.